### PR TITLE
cargo-fmt: detect Rust edition in use

### PR DESCRIPTION
**Pull request by otavio**

### CODE REVIEW

File: src/cargo-fmt/main.rs [hunk 1]
```diff
@@ -334,41 +318,41 @@ fn add_targets(target_paths: &[cargo_metadata::Target], targets: &mut HashSet<Ta
...
-    files: &[PathBuf],
+    targets: &HashSet<Target>,
     fmt_args: &[String],
     verbosity: Verbosity,
-) -> Result<ExitStatus, io::Error> {
-    let stdout = if verbosity == Verbosity::Quiet {
-        std::process::Stdio::null()
-    } else {
-        std::process::Stdio::inherit()
-    };
```
Comment (otavio) [inline src/cargo-fmt/main.rs]: This is something I don't know how to handle as I cannot inherit it and use it on multiple calls for `Command` as it is consumed. Does anyone have a hint on how to deal with this?

**Comment by otavio** [comment]
Looking at the build issues, they seem to not be related to this change. In fact, it seems `cargo` needs to have its code formatted using the latest version of `rustfmt` as it had a few significant changes.

File: src/cargo-fmt/main.rs [hunk 2]
```diff
@@ -334,41 +318,48 @@ fn add_targets(target_paths: &[cargo_metadata::Target], targets: &mut HashSet<Ta
...
+        .map(|t| (&t.edition, vec![&t.path]))
+        .collect();
 
-    if verbosity == Verbosity::Verbose {
-        print!("rustfmt");
-        for a in fmt_args {
-            print!(" {}", a);
-        }
-        for f in files {
-            print!(" {}", f.display());
```
Comment (nrc) [inline src/cargo-fmt/main.rs]: I think we should keep this stuff.

File: src/cargo-fmt/main.rs [hunk 3]
```diff
@@ -334,41 +318,48 @@ fn add_targets(target_paths: &[cargo_metadata::Target], targets: &mut HashSet<Ta
...
+                    "Could not run rustfmt, please make sure it is in your PATH.",
+                ),
+                _ => e,
+            })?;
+
+        let status = command.wait()?;
+        if !status.success() {
+            return Ok(status.code().unwrap_or(FAILURE));
         }
-        println!();
```
Comment (nrc) [inline src/cargo-fmt/main.rs]: We need to keep this or the output looks bad

**Comment by otavio** [comment]
@nrc thanks for reviewing it. I ended forgetting to add it back.

Please take a look at it now.

**Comment by nrc** [comment]
Thank you!

**Pull request merged**